components/brave_wallet/browser/bitcoin/bitcoin_wallet_service.cc

```diff
@@ -452,6 +478,35 @@ void CreateTransactionTask::WorkOnTask() {
     return;
   }
 
+  // Guard against requests that exceed the available balance early to avoid
+  // constructing invalid transactions.
+  if (!transaction_.sending_max_amount()) {
+    base::CheckedNumeric<uint64_t> available = 0;
+    for (const auto& [_, utxos] : utxo_map_) {
+      for (const auto& utxo : utxos) {
+        uint64_t utxo_value = 0;
+        if (!base::StringToUint64(utxo.value, &utxo_value)) {
+          SetError(WalletInternalErrorMessage());
+          ScheduleWorkOnTask();
+          return;
+        }
+
+        available += utxo_value;
+        if (!available.IsValid()) {
+          SetError(WalletInternalErrorMessage());
+          ScheduleWorkOnTask();
+          return;
+        }
+      }
+    }
+
+    if (transaction_.amount() > available.ValueOrDie()) {
+      SetError(WalletInsufficientBalanceErrorMessage());
+      ScheduleWorkOnTask();
+      return;
+    }
+  }
+
```

Comment:
This naturally should part of knapsack solver. Like try fail fast when randomly picked utxos will never reach send amount.

Thats OK to fail after DiscoverNextUnusedAddress and GetFeeEstimates RPC requests as acutally UI also validates wallet amount vs send amount and we are not supposed to hit this validation frequently